modules/configuring-hpa-based-on-application-metrics.adoc

```diff
@@ -0,0 +1,44 @@
+// Module included in the following assemblies:
+//
+// * machine_management/configuring-hpa-for-an-application.adoc
+
+[id='configuring-hpa-based-on-application-metrics-{context}']
+= Configuring HPA based on application metrics
+
+If you have configured an application to export metrics, you can set up Horizontal Pod Autoscaling (HPA) based on these metrics.
+
+.Procedure
+
+. Create a YAML file for your configuration. In this example, it is called `deploy.yaml`.
+
+. Add configuration for deploying the horizontal pod autoscaler for the application. This example configures and deploys HPA based on the application `http_requests_per_second` metric for the sample application configured in the "Application monitoring" section:
++
+[source,yaml]
+----
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: example-app-scaler
+  namespace: default
+spec:
+  scaleTargetRef:
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    name: example-app <1>
+  minReplicas: 3 <2>
+  maxReplicas: 10 <3>
+  metrics:
+  - type: Pods
+    pods:
+      metricName: http_requests_per_second <4>
+      targetAverageValue: 10 <5>
+----
+<1> `name` specifies the application
+<2> `minReplicas` specifies the minimum number of replicas the HPA should maintain for the application
+<3> `maxReplicas` specifies the maximum number of replicas the HPA should maintain for the application
+<4> `metricName` specifies the metric upon which HPA is based
```

Comment:
Can you use any valid metric in the cluster, or are there limits? Please clarify.